src/Payment.ts

```diff
@@ -95,12 +95,17 @@ export class Payment {
     networkAddress: string,
     receiverAddress: string,
     value: number | string,
-    options: PaymentOptions = {}
+    options: PaymentOptions = {},
+    referenceId: string = ''
   ): Promise<PaymentTxObject> {
     const { gasPrice, gasLimit, networkDecimals, extraData } = options
     const decimals = await this.currencyNetwork.getDecimals(networkAddress, {
       networkDecimals
     })
+    const extraDataWithReferenceId = this.extendExtraDataByReferenceId(
```

Comment:
A more generic name would be more convenient when we will need to add more in the future, like `encodeExtraData`

Reply:
Hmm I'm not sure what this means. Is this then a generic function which takes an extraData string plus a dictionary of all possible values to set? So far it is a very focused function to just encodes the given reference. So since the encoding is totally unclear at the moment, I would not know how to rework this function without making assumptions for the signature. While the current one do no assumptions at all.
If you have a concrete idea in mind and I just stuck here, please don't hesitate to make a suggestion or add a commit by yourself. I would be glad if so.